web/src/zform.ts

```diff
@@ -8,22 +8,17 @@ import type {Message} from "./message_store.ts";
 import type {Event} from "./poll_widget.ts";
 import {zform_widget_extra_data_schema} from "./submessage.ts";
 import * as transmit from "./transmit.ts";
+import type {WidgetExtraData} from "./widgetize.ts";
 
 export type ZFormExtraData = z.infer<typeof zform_widget_extra_data_schema>;
 
 export function activate(opts: {
     $elem: JQuery;
-    extra_data: unknown;
+    extra_data: WidgetExtraData;
     message: Message;
-}): {handle_events: (events: Event[]) => void} | undefined {
+}): (events: Event[]) => void {
     const $outer_elem = opts.$elem;
-    const parse_result = zform_widget_extra_data_schema.safeParse(opts.extra_data);
-
-    if (!parse_result.success) {
-        blueslip.warn("invalid zform extra data", {issues: parse_result.error.issues});
-        return undefined;
-    }
-    const {data} = parse_result;
+    const data = zform_widget_extra_data_schema.parse(opts.extra_data);
```

Comment:
Removing this safeParse code seems like a material behavior change. What's the motivation for it?

Reply:
By adding types, it became clear that it was already parsed with a known type by the time it got here, in `submessage` and the setup to `widgetize.activate`. I can add that to the commit message.